website/src/developing-extensions/custom-deprecations.md

```diff
@@ -0,0 +1,59 @@
+---
+title: Custom deprecations
```

Comment:
This is a new page. Needs to be referenced in https://github.com/phpstan/phpstan/blob/2.1.x/website/src/developing-extensions/developing-extensions.json.